core/src/main/java/org/apache/accumulo/core/gc/ReferenceFile.java

```diff
@@ -49,16 +67,16 @@ public TableId getTableId() {
   }
 
   @Override
-  public String getMetadataEntry() {
-    return metadataEntry;
+  public String getMetadataPath() {
```

Comment:
Looking at the javadoc for [Reference.getMetadataPath()](https://github.com/apache/accumulo/blob/f501dca59ae47179c6f62a134e44b27a3d34f652/core/src/main/java/org/apache/accumulo/core/gc/Reference.java#L39-L45) it writes about this data being exactly what is stored in the metadata table.  That is not technically true, what it stored in the metadata table is the json.  However I don't think it needs to be exactly what is stored in the metadata table.  Need to figure out if there is need for the exact metadata info or if the the javadoc needs to be updated.   I suspect the javadoc needs to be updated, but not completely sure.  Could open a follow on issue to investigate this further.

Reply:
I think the javadoc here needs to be updated and I can do that. I made changes and forgot to update it. The path here could be exactly what's in metadata (such as if it's a `StoredTabletFile`) or may be normalized if the reference is `ScanServerRefTabletFile`.

I don't think it actually needs to be the exact path since we are not using it to remove entries in the metadata table and where the path is used later by `GarbageCollectionAlgorithm` it is actually processed and made relative anyways [here](https://github.com/cshannon/accumulo/blob/ecf7c7ac31379b5e5fc95a33df4237927edd8a9c/server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionAlgorithm.java#L175)

Reply:
> I don't think it actually needs to be the exact path since we are not using it to remove entries in the metadata table and where the path is used later by GarbageCollectionAlgorithm it is actually processed and made relative anyways [here](https://github.com/cshannon/accumulo/blob/ecf7c7ac31379b5e5fc95a33df4237927edd8a9c/server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionAlgorithm.java#L175)

I agree with that and saw the same thing when I looked.  Based on that it does not seem we need the exact bytes that are stored in the metadata table.